pkg/config/nodetreemodel/getter_test.go

```diff
@@ -0,0 +1,230 @@
+	cfg.SetDefault("a", []string{"a", "b", "c"})
+	cfg.SetDefault("b", "a b c")
+	cfg.BuildSchema()
+
+	assert.Equal(t, []string{"a", "b", "c"}, cfg.GetStringSlice("a"))
+	assert.Equal(t, []string{"a", "b", "c"}, cfg.GetStringSlice("b"))
+	assert.Equal(t, []string(nil), cfg.GetStringSlice("does_not_exists"))
+}
+
+func TestGetStringMap(t *testing.T) {
+	cfg := NewConfig("test", "", nil)
+	cfg.SetDefault("a", map[string]interface{}{"a": 1, "b": "b", "c": nil})
+	cfg.SetDefault("b", "{\"a\": 1234}") // viper handles JSON string implicitly so we have to reproduce this behavior
+	cfg.BuildSchema()
+
+	assert.Equal(t, map[string]interface{}{"a": 1, "b": "b", "c": nil}, cfg.GetStringMap("a"))
+	assert.Equal(t, map[string]interface{}{"a": 1234.0}, cfg.GetStringMap("b"))
+	assert.Equal(t, map[string]interface{}{}, cfg.GetStringMap("does_not_exists"))
+}
+
+func TestGetStringMapString(t *testing.T) {
+	cfg := NewConfig("test", "", nil)
+	cfg.SetDefault("a", map[string]interface{}{"a": 123, "b": "b", "c": ""})
+	cfg.SetDefault("b", "{\"a\": \"test\"}") // viper handles JSON string implicitly so we have to reproduce this behavior
+	cfg.BuildSchema()
+
+	assert.Equal(t, map[string]string{"a": "123", "b": "b", "c": ""}, cfg.GetStringMapString("a"))
+	assert.Equal(t, map[string]string{"a": "test"}, cfg.GetStringMapString("b"))
+	assert.Equal(t, map[string]string{}, cfg.GetStringMapString("does_not_exists"))
+}
+
+func TestGetStringMapStringSlice(t *testing.T) {
+	cfg := NewConfig("test", "", nil)
+	cfg.SetDefault("a", map[string][]interface{}{"a": {1, 2}, "b": {"b", "bb"}, "c": nil})
+	cfg.SetDefault("b", "{\"a\": [\"test\", \"test2\"]}") // viper handles JSON string implicitly so we have to reproduce this behavior
+	cfg.BuildSchema()
+
+	assert.Equal(t, map[string][]string{"a": {"1", "2"}, "b": {"b", "bb"}, "c": nil}, cfg.GetStringMapStringSlice("a"))
+	assert.Equal(t, map[string][]string{"a": {"test", "test2"}}, cfg.GetStringMapStringSlice("b"))
+	assert.Equal(t, map[string][]string{}, cfg.GetStringMapStringSlice("does_not_exists"))
+}
+
+func TestGetSizeInBytes(t *testing.T) {
+	cfg := NewConfig("test", "", nil)
+	cfg.SetDefault("a", "123")
+	cfg.SetDefault("b", "1kb")
+	cfg.SetDefault("c", "1Mb")
+	cfg.SetDefault("d", "1 GB")
+	cfg.SetDefault("e", "-1")
+	cfg.SetDefault("f", "invalid")
+	cfg.BuildSchema()
+
+	assert.Equal(t, uint(123), cfg.GetSizeInBytes("a"))
+	assert.Equal(t, uint(1024), cfg.GetSizeInBytes("b"))
+	assert.Equal(t, uint(1024*1024), cfg.GetSizeInBytes("c"))
+	assert.Equal(t, uint(1024*1024*1024), cfg.GetSizeInBytes("d"))
+	assert.Equal(t, uint(0), cfg.GetSizeInBytes("e"))
+	assert.Equal(t, uint(0), cfg.GetSizeInBytes("f"))
+}
+
+func TestGetFloat64Slice(t *testing.T) {
+	cfg := NewConfig("test", "", nil)
+	cfg.SetDefault("float_list", []float64{})
+	cfg.SetDefault("string_list", []string{})
+	cfg.BuildSchema()
+	cfg.ReadConfig(strings.NewReader(`---
+float_list:
+  - 1.1
+  - "2.2"
+  - 3
+string_list:
+  - 1.1
+  - "2.2"
+  - 3
```

Comment:
These two cases seem identical. Is this intentional?

Reply:
The default is different. On default to `[]string` and the other `[]float64`. For now, defaults have no impact but it will in the future so this test cover the different usage from dogstatsd.